### ISSUE TITLE
MTP/USB file transfer from Windows PC disconnects constantly on Pixel 8 Pro

### ISSUE THREAD

**Issue by Doc-4** [GrapheneOS/os-issue-tracker#3125]
This sounds oddly similar but distinct to this issue
https://github.com/GrapheneOS/os-issue-tracker/issues/2611

After anywhere between 10 seconds and a little over a minute of transferring files from my windows PC to my Pixel 8 Pro over USB with the stock cable (and others) using multiple different USB ports, the process hangs and crashes as if the device was disconnected. The device will be there momentarily afterwards, it's like a short total disconnect. Affects file transfers with both File Explorer and 3rd party programs. Tried this ad nauseam, tested other android devices on the same computer, transferring large files. They work, pixel doesn't. Used a 3rd party app on the pixel to transfer files via FTP in the end, that works fine. MTP file transfer between android to android also works. I can test this further if that would be helpful.

**Comment by thestinger** [comment]
Seems to be an issue with the computer. You should try using MTP from another phone or computer to confirm it works properly.

**Comment by Doc-4** [comment]
> Seems to be an issue with the computer. You should try using MTP from another phone or computer to confirm it works properly.

I'm sorry if I wasn't clear in the initial report, but I did test this. I tested it on my last phone, a Oneplus 7T Pro. Same file transfer, same cable, same USB port, same computer. No issues when done with that phone.

**Comment by thestinger** [comment]
I mean you need to test the Pixel 8 Pro with another computer or phone using the MTP it provides.

**Comment by Doc-4** [comment]
> I mean you need to test the Pixel 8 Pro with another computer or phone using the MTP it provides.

Okay, I'll do that.

**Comment by Doc-4** [comment]
> I mean you need to test the Pixel 8 Pro with another computer or phone using the MTP it provides.

I've tested this further and I think it might be a specific interaction between Windows itself and programs, the MTP USB driver, and one of the differing qualities about GrapheneOS Android and my last phone such as just being newer with Android 14 or being 64-bit only. The bug persists with other programs later on but doesn't seem to be able to be caused by those programs. Sorry for the false flag, I read that other report and it sounded like the same symptom.

**Comment by thestinger** [comment]
It might help someone else in the future. It seems like a Windows bug, driver bug, hardware bug on the desktop side, or some kind of software interfering.

**Comment by AMDBartek** [comment]
Can actually confirm this issue on Kubuntu 23.10. MTP frequently just breaks on my Pixel 8 Pro running GrapheneOS and requires a replug to get working again. Then breaks again after transferring something like just 2 small files.

**Comment by Doc-4** [comment]
I've actually found that this issue persists across other devices than just windows PC since, it's just inconsistent. The stock cable definitely aggravates it to be worse than it already is, bending it can cause the device to disconnect and reconnect on the same PC port that otherwise works fine. Issue still exists otherwise even with a better cable, just less common. It tends to happen to me far more often with a large operation than just a single file transfer, syncing music would almost 100% of the time result in a failure mid way, but usually in the first minute

**Comment by AMDBartek** [comment]
Yes, large operations exaggerate the effect, however it is not a cable/connection issue from what I can see. Bending the cable has no effect and I used a cable tester on my cable to verify this. It looks to me like a software bug...

**Comment by AMDBartek** [comment]
@thestinger Must be something upstream then? Because I literally did a backup of my phone via MTP to move to GrapheneOS soo... it cannot be a stock issue

**Comment by Doc-4** [comment]
It may not be the cable but rather something in software that makes Windows think the device was disconnected. I'm going by the little disconnect sound you get from the operating system. I believe this to be a software issue as well, I'm just listing examples to show I tested various hardware to eliminate them as the cause.

**Comment by Doc-4** [comment]
Yeah if any system logs or tests would be helpful, happy to help in determining the source as well

**Comment by AMDBartek** [comment]
Firstly, I want to apologise about whatever rule I broke and I am sorry if this comment also breaks the rules. I am writing this with the good intentions and I do not want to break any rule that I may not be aware of so please feel free to let me know where I can read these rules so I am informed in the future, but I want to point out that I fixed this issue by simply blowing air into my Pixel 8 Pro's USB-C port and my cable. This is not an issue with GrapheneOS in my specific case.

**Comment by AMDBartek** [comment]
@thestinger Wow. I truly feel bad about this whole situation and wish you and every member of the GrapheneOS team all of the best. It is *completely* unacceptable to attack someone in such severe ways like that over something digital like an operating system.

And once again, sorry. I really didn't mean to cause such a big hassle. I was not aware of the situation between various communities.

**Comment by Doc-4** [comment]
I'm sorry if I had broken a rule, I've reviewed them and while I didn't bring up any other OS, I suppose it could be "Sustained disruption or derailing of threads, channels, lists, and similar forums", or caught up in perceived trolling, I see it and know it's unfortunately common, so to think that is understandable. It's not my intention to derail anything or participate in any reductive manner either way, so if I said something unhelpful, I'm sorry. 

On the MTP issue, I did try blowing out the USB-C contacts, and had no issues until I tried to access the device from more than one program without disconnecting it. If I used file explorer successfully or the media center application first, they work. When I use one after another, in sequence, after going from the second program back to the first, the device through file explorer reported "phantom" files that I had already deleted, and would disappear on reconnect later. At this point, file explorer locked up and crashed, and upon restart the device wasn't visible as it should be, with device manager reporting a malfunction of a USB device. Specifically:

"Windows has stopped this device because it has reported problems (Code 43)

The USB set address request failed"

I'm not going to make any assumptions as to what this means, for all I know the program I use for music has a bug on newer android versions and caused the malfunction, and it has nothing to do with GOS or android MTP. I'm just trying to report this information accurately in hopes it will be helpful in determining the cause.
